opesci/staggeredgrid.py

```diff
@@ -903,24 +990,36 @@ def stress_bc(self):
                         # loop through other dimensions
                         if not d2 == d:
                             i = self.index[d2]
-                            i0 = 0
-                            i1 = self.dim[d2]
+                            if normal:
+                                if field.direction[0] == d+1:
+                                    # Txx in x plane
+                                    i0 = 0
+                                    i1 = self.dim[d2]
+                                else:
+                                    # Txx in y, z plane
+                                    i0 = self.margin.value+1
+                                    i1 = self.dim[d2]-self.margin.value-1
+                            else:
+                                i0 = 0
+                                i1 = self.dim[d2]
                             if body == '':
                                 # inner loop, populate ghost cell calculation
-                                body = field.bc[d+1][side]
+                                # body = field.bc[d+1][side]
+                                if self.read:
+                                    body = ''.join(ccode_eq(self.resolve_media_params(bc))+';\n' for bc in field.bc[d+1][side])
+                                else:
+                                    body = ''.join(ccode_eq(bc)+';\n' for bc in field.bc[d+1][side])
                                 dict1 = {'i': i, 'i0': i0,
                                          'i1': i1, 'body': body}
-                                body = render(tmpl, dict1).replace('[t]',
-                                                                   '[t1]')
+                                body = render(tmpl, dict1).replace('[t + 1]', '[t1]').replace('[t]', '[t0]')
                                 if self.ivdep:
-                                    body = '%s\n' % self.compiler._ivdep + body
+                                    body = '#pragma ivdep\n' + body
```

Comment:
The `ivdep` pragma is compiler-dependent, ie. GCC requires it to be `#pragma GCC ivdep`, so we should get this from the compiler object.